pkg/compose/hash.go

```diff
@@ -40,3 +41,23 @@ func ServiceHash(o types.ServiceConfig) (string, error) {
 	}
 	return digest.SHA256.FromBytes(bytes).Encoded(), nil
 }
+
+// From a top-level Volume Configuration, creates a unique hash ignoring
+// DriverOpts,  External and Extensions
+func VolumeHash(o types.VolumeConfig) (string, error) {
+	// We can't have these attributes into account, because when converting
+	// from a volume inspect there is no way to deduce these attributes back
+	// to a VolumeConfig
+	if o.Driver == "" { // (TODO: jhrotko) This probably should be fixed in compose-go
+		o.Driver = "local"
+	}
+	o.External = false // the name can change. Need to think about this case
+	o.Labels = nil
```

Comment:
why drop labels ?

Reply:
can you add labels in a compose configuration?

Reply:
sure: https://github.com/compose-spec/compose-spec/blob/main/07-volumes.md#labels
Remember compose `volumes` is just a 1:1 mapping for `docker volume create`.